pkg/ipam/core/ipam.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipamcore
+
+import (
+	"fmt"
+	"net/netip"
+	"slices"
+)
+
+// Ipam represents the IPAM core structure.
+type Ipam struct {
+	roots []node
+}
+
+// NewIpam creates a new IPAM instance.
+func NewIpam(roots, preallocated []string) (*Ipam, error) {
+	ipamRootsPrefixes := make([]netip.Prefix, len(roots))
+	for i, root := range roots {
+		ipamRootsPrefixes[i] = netip.MustParsePrefix(root)
+	}
+
+	ipamPreallocated := make([]netip.Prefix, len(preallocated))
+	for i, prefix := range preallocated {
+		ipamPreallocated[i] = netip.MustParsePrefix(prefix)
+	}
```

Comment:
I think this feature it is not needed anymore with https://github.com/liqotech/liqo/pull/2829, as the reserved networks (podcidr, servicecidr) are allocated with grpc calls before starting controllers